File: mongodb/net9/Services/Test1Service.cs
using MongoDB.Bson;
using MongoDB.Driver;
using net9.Context;
using net9.Models.Entities;
using net9.Models.Requests;
using net9.Models.Responses;

namespace net9.Services;

public class Test1Service: ITest1Service
{
    private readonly IMongoCollection<Test1> _test1;

    public Test1Service(MongoDbContext mongoDbContext)
    {
        _test1 = mongoDbContext.Test1;
    }

    public async Task<Response<CreateResponse>> Create(CreateRequest createRequest)
    {
        try
        {
            var test = new Test1 {Test = createRequest.Test};
            await _test1.InsertOneAsync(test);
            return ResponseHelper.SetCreatedResponse(new CreateResponse { Id = test.Id.ToString(), Test = test.Test });
        }
        catch(Exception e)
        {
            Console.WriteLine($"error: {e}");
            return ResponseHelper.SetInternalServerErrorResponse<CreateResponse>();
        }
    }

    public async Task<Response<GetByIdResponse>> GetById(string id)
    {
        try
        {
            if (!ObjectId.TryParse(id, out var objectId)) {
                return ResponseHelper.SetBadRequestResponse<GetByIdResponse>("id is not a objectId");
            }
            var test1 = await _test1.Find(m => m.Id == objectId).FirstOrDefaultAsync();
            if (test1 == null) {
                return ResponseHelper.SetNotFoundResponse<GetByIdResponse>($"cannot find test1 with id: {id}");
            }
            return ResponseHelper.SetOkResponse(new GetByIdResponse { Id = test1.Id.ToString(), Test = test1.Test });
        }
        catch(Exception e)
        {
            Console.WriteLine($"error: {e}");
            return ResponseHelper.SetInternalServerErrorResponse<GetByIdResponse>();
        }
    }

    public async Task<Response<UpdateResponse>> Update(UpdateRequest updateRequest)
    {
        try
        {
            if (!ObjectId.TryParse(updateRequest.Id, out var objectId)) {
                return ResponseHelper.SetBadRequestResponse<UpdateResponse>("id is not objectid");
            }
            var test1 = await _test1.Find(m => m.Id == objectId).FirstOrDefaultAsync();
            if (test1 == null) {
                return ResponseHelper.SetNotFoundResponse<UpdateResponse>($"cannot find test1 with id: {updateRequest.Id}");
            }
            var updatedTest1 = Builders<Test1>.Update
                .Set(m => m.Test, updateRequest.Test);
            var result = await _test1.UpdateOneAsync(m => m.Id == objectId, updatedTest1);
            if (result.ModifiedCount == 0) {
                return ResponseHelper.SetNotFoundResponse<UpdateResponse>($"there is no updated data");
            }
            return ResponseHelper.SetOkResponse(new UpdateResponse{Id = updateRequest.Id, Test = updateRequest.Test});
        }
        catch(Exception e)
        {
            Console.WriteLine($"error: {e}");
            return ResponseHelper.SetInternalServerErrorResponse<UpdateResponse>();
        }
    }

    public async Task<Response<MessageResponse?>> Delete(DeleteRequest deleteRequest)
    {
        try
        {
            if (!ObjectId.TryParse(deleteRequest.Id, out var objectId)) {
                return ResponseHelper.SetBadRequestResponse<MessageResponse?>("id is not objectid");
            }
            var test1 = await _test1.Find(m => m.Id == objectId).FirstOrDefaultAsync();
            if (test1 == null) {
                return ResponseHelper.SetNotFoundResponse<MessageResponse?>($"cannot find test1 with id: {deleteRequest.Id}");
            }
            var result = await _test1.DeleteOneAsync(m => m.Id == objectId);
            if (result.DeletedCount == 0) {
                return ResponseHelper.SetNotFoundResponse<MessageResponse?>($"there is no updated data");
            }
            return ResponseHelper.SetNoContentResponse<MessageResponse>(null);
        }
        catch(Exception e)
        {
            Console.WriteLine($"error: {e}");
            return ResponseHelper.SetInternalServerErrorResponse<MessageResponse?>();
        }
    }
}

File: postgresql/nextjs/src/services/test1-service.ts
import { Test1 } from "../models/entities/test1";
import { CreateRequest } from "../models/requests/create-request";
import { DeleteRequest } from "../models/requests/delete-request";
import { UpdateRequest } from "../models/requests/update-request";
import { CreateResponse } from "../models/responses/create-response";
import { GetByIdResponse } from "../models/responses/get-by-id-response";
import { MessageResponse, Response, setInternalServerErrorResponse, setNoContentResponse, setNotFoundResponse, setOkResponse } from "../models/responses/response";
import { UpdateResponse } from "../models/responses/update-response";
import { Test1Repository } from "../repositories/test1-repository";
import { pool } from "../utils/postgres-util";

export class Test1Service {
    static async create(createRequest: CreateRequest): Promise<Response<CreateResponse | null>> {
        const client = await pool.connect()
        try {
            await client.query("BEGIN")
            const test1: Test1 = {
                test: createRequest.test
            }
            const result = await Test1Repository.create(client, test1)
            if (result.rowCount != 1) {
                return setInternalServerErrorResponse()
            }
            test1.id = result.rows[0].id
            await client.query("COMMIT")
            const createResponse: CreateResponse = {
                id: test1.id,
                test: test1.test
            }
            return setOkResponse(createResponse)
        } catch(e) {
            await client.query("ROLLBACK")
            console.log("error:", e);
            return setInternalServerErrorResponse()
        } finally {
            if (client) {
                client.release()
            }
        }
    }

    static async getById(id: number): Promise<Response<GetByIdResponse | null>> {
        try {
            const client = await pool.connect()
            const result = await Test1Repository.getById(client, id)
            if (result.rowCount === 0) {
                return setNotFoundResponse("cannot find test1 with id: " + id)
            }
            const getByIdResponse: GetByIdResponse = {
                id: result.rows[0].id,
                test: result.rows[0].test
            }
            return setOkResponse(getByIdResponse)
        } catch(e) {
            console.log("errror:", e)
            return setInternalServerErrorResponse()
        }
    }

    static async update(updateRequest: UpdateRequest): Promise<Response<UpdateResponse | null>> {
        const client = await pool.connect()
        try {
            await client.query("BEGIN")
            const test1: Test1 = {
                id: updateRequest.id,
                test: updateRequest.test
            }
            const result = await Test1Repository.update(client, test1)
            if (result.rowCount != 1) {
                return setInternalServerErrorResponse()
            }
            await client.query("COMMIT")
            const updateResponse: UpdateResponse = {
                id: test1.id,
                test: test1.test
            }
            return setOkResponse(updateResponse)
        } catch(e) {
            await client.query("ROLLBACK")
            console.log("error:", e)
            return setInternalServerErrorResponse()
        } finally {
            if (client) {
                client.release()
            }
        }
    }

    static async delete(deleteRequest: DeleteRequest): Promise<Response<MessageResponse | null>> {
        const client = await pool.connect()
        try {
            await client.query("BEGIN")
            const result = await Test1Repository.delete(client, deleteRequest.id)
            if (result.rowCount != 1) {
                return setInternalServerErrorResponse()
            }
            await client.query("COMMIT")
            return setNoContentResponse(null)
        } catch(e) {
            await client.query("ROLLBACK")
            console.log("error:",e)
            return setInternalServerErrorResponse()
        } finally {
            if (client) {
                client.release()
            }
        }
    }
}